--- src/init.c.orig	2019-07-28 10:49:43.320029678 -0500
+++ src/init.c	2019-07-28 10:50:00.354048664 -0500
@@ -295,7 +295,7 @@
 #endif /* USE_PRIVATE_PATHS */
 
 	/* Build the path to the score, save and archive directories */
-	BUILD_DIRECTORY_PATH(ANGBAND_DIR_SCORES, userpath, "scores");
+	BUILD_DIRECTORY_PATH(ANGBAND_DIR_SCORES, datapath, "scores");
 	BUILD_DIRECTORY_PATH(ANGBAND_DIR_SAVE, userpath, "save");
 	BUILD_DIRECTORY_PATH(ANGBAND_DIR_ARCHIVE, userpath, "archive");
 
